### PR TITLE
use IPv6 port binding

### PR DESCRIPTION
**Pull request by fabianfrz**
Change listening IP to :: and network stack to IPv6.

On linux, this does not make a difference if the user has only IPv4 since we don't set the flag to disable multi stack support (IPv6 bindings are backward compatible).

Note: I applied those changes from my local installation so I have not tested it with this exact branch.  The debian package says that it is "Version: 1.2.0"

### CODE REVIEW

**Comment by jketterl** [comment]
As far as I'm informed, the IPv6 compatibility in the Python webserver has only been established in Python 3.8 (see https://groups.io/g/openwebrx/topic/76531376 for previous discussion). Since this project currently aims to support Python versions starting with 3.5 (the version contained in Debian Stretch), I cannot include this change at the current point in time.

If you have different information, please let me know.

File: owrx/socket.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 
 def getAvailablePort():
-    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
```
Comment (jketterl) [inline owrx/socket.py:5]: This method has nothing to do with the webserver, it is used for internal communication. I'm not sure if all affected tools support IPv6 yet.

Comment (fabianfrz) [reply owrx/socket.py:5]: I will check if it works without this change. If yes, I will revert this line.

**Comment by fabianfrz** [comment]
Hi @jketterl, I would like thank you for your prompt reply. I installed openwebrx on my raspberry pi (->raspbian) which is running on bullseye. This version provides Python 3.9.

I don't know what stretch is currently using. However you could consider it as dead since June:
https://wiki.debian.org/LTS

**Comment by jketterl** [comment]
Yes, but even then there's still only version 3.7 in Debian Buster.

**Comment by fabianfrz** [comment]
Then this might cause a problem if it should receive updates and be managed by a deb file. For old distributions, there are workarounds like SCL on RHEL however I don't know a easy way beside full containerization (Docker) or installing a custom python interpreter on Debian.

**Comment by jketterl** [comment]
Not sure why .deb packaging should be a problem... There has been .deb packages of this software for some time now, and things have been working fine.

The bug that has been in Python for so long isn't really helpful because in effect, it only allows you to use either IPv4 or IPv6, and given that IPv4 is far from being on its way out, it's simply too early to switch completely. It's not my fault that this has been in a major distribution for so long, and it's also not my fault that not everybody uses rolling distributions. This is just a case of a transition that can only happen at the best pace of all those involved.

The use of a reverse proxy is a common practice, and for now, I'd say it's the recommended way for getting IPv6 support in this case.

**Comment by fabianfrz** [comment]
> Not sure why .deb packaging should be a problem... There has been .deb packages of this software for some time now, and things have been working fine.

Sorry, wrote it a bit confused. Debian based distributions do not provide multiple versions of the same software like in this case multiple python versions where you can choose from.

If it is too early for this change, I have to apply the patch again and again until it is finally merged but after all, this is your repo.

**Comment by fabianfrz** [comment]
@jketterl actually there is a packaging mechanism for this. I guess the debian directory needs a patches directory and then the diff files needs to be put in there. At least it looks like this in the [ZFS project](https://salsa.debian.org/zfsonlinux-team/zfs/-/tree/master/debian).

If you don't know it - you can append .pach or .diff to supporting GitHub URLs to get that format out of a merge request.
However putting the patch blindly into the debian/patches directory is also a bad idea since this MR only needs to be reverted for buster but not for bullseye.

**Comment by jketterl** [comment]
OK so this should hopefully be covered with the latest commits in "develop" now. I have decided to simply make it a configurable core config option for now. The default is enabled because most recent versions of python should make it run as a dual-stack solution that way. Users that still run older python versions should be able to disable IPv6 support that way, resorting to an IPv4 only solution, as it was previously.

I did look into solutions to include patches during the packaging process, but the packages are built using the "native" format (as opposed to "quilt") which simply does not have that option.

**Comment by fabianfrz** [comment]
going that way is also a good way to go. maybe even better in some setups.

**Pull request closed without merging**